### DISCUSSION TITLE
How to convert *stats.hsol

### DISCUSSION THREAD

**Discussion by Dewu-Yang** [loganoz/horses3d#164] [Q&A] [answered]
Dear prof. Gonzalo

      I am interested in HORSES3D as well, and I have been studying it recently. I have encountered a question that I would like to ask you. How can I open or convert the statistical file .stats.hsol? I tried using the command ./horses2plt *.stats.hsol *.hmsh to convert it to a .tec file, but it didn't work. I have looked at the user-guide documentation, but I couldn't find any information on how to open or convert the .stats.hsol file. I would greatly appreciate it if you could provide some suggestions. I am looking forward to your response. Thanks!

Best,
Dewu

**Comment by oscarmarino** [comment]
Dear Dewu,

Converting stats.hsol files to tec works similar to normal (instantanous) hsol files. You need to specify the actual file (with correct relative or absolute path) and whether you want some specifics, i.e. variables, p order, etc. This is an example running it from the RESULTS directory:
./horses2plt case_name.stats.hsol ../MESH/case_name.hmesh --dimensionless --output-mode=FE --output-variables=Sxx,Syy,Szz,Vmean

Did you get an specific error?

Best regards,
Oscar

**Reply by Dewu-Yang** [reply]
Dear Oscar,

 Thank you very much for your response. Yes, I tried again following the command you provided, and I encountered the same error. The error message is as follows: 
_Array found in file dimensions does not match that of the introduced variable. File dimensions:    -33233683 , Variable:            4
Error in file SolutionFile.f90, in line 618._
How to solve this error? Thank you again for your response. I look forward to your suggestions once again. Thanks!

Best,
Dewu

**Comment by oscarmarino** [comment, marked as answer]
Hi,

Seems that there are more variables in the hsol that the standard. Are you saving the gradients in the simulation?

If that is the case, you can use a control file, similar to this:
hmesh file   = "../MESH/case_name.hmesh"
hsol file = "./case_name.hsol"
output variables = Sxx,Syy,Szz,Vmean
output mode = FE
dimensionless = .true.
has gradients = .true.

and run it like:
/horses2plt file_name.control from the RESULTS directory.

Hope it helps.

Best regards,
Oscar

**Reply by Dewu-Yang** [reply]
Dear Oscar,

Thank you very much for your suggestion. I tried the method you provided, and it worked. I greatly appreciate your clarification.

Best,
Dewu
